perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ExtSourcesManagerBlImpl.java

```diff
@@ -281,11 +279,12 @@ public Candidate getCandidate(PerunSession perunSession, Map<String,String> subj
 				attributes.put(attrName, subjectData.get(attrName));
 			}
 		}
-		List<UserExtSource> additionalUserExtSources = Utils.extractAdditionalUserExtSources(perunSession, subjectData);
-		candidate.setAdditionalUserExtSources(additionalUserExtSources);
-		candidate.setAttributes(attributes);
 
-		return candidate;
+		candidateSync.setRichUserExtSource(new RichUserExtSource(userExtSource, new ArrayList<>()));
```

Comment:
I am just wondering about a use case where we want to find users by some attribute from the main userExtSource. I think this that the way the code is written now, it will be not possible. What do you think @metodej and @stavamichal ?

Reply:
I am not sure where would I find the attributes for main ues... Could they be part of the subjectData like the member and user attributes?

Reply:
At this moment there is no way how to set attributes for main extSource. It would probably need another iteration (other PR) to fix this specific problem. You can't fix it here.